pkg/controller/sparkapplication/controller.go

```diff
@@ -303,7 +303,7 @@ func (c *Controller) getUpdatedAppStatus(app *v1alpha1.SparkApplication) v1alpha
 	}
 
 	if currentDriverState != nil {
-		if newAppState, err := driverPodPhaseToApplicationState(currentDriverState.podPhase); err == nil {
+		if newAppState := driverPodPhaseToApplicationState(currentDriverState.podPhase); newAppState != v1alpha1.UnknownState {
```

Comment:
You aren't setting the appStatus anywhere to Unknown.

Reply:
Done.